packages/react/src/file-uploader/FileUploader.tsx

```diff
@@ -0,0 +1,31 @@
+import { FileUpload } from "@ark-ui/react";
+
+export const FileUploader = () => {
+  return (
+    <FileUpload.Root maxFiles={5}>
+      <FileUpload.Label>File Upload</FileUpload.Label>
+      <FileUpload.Dropzone>Drag your file(s) here</FileUpload.Dropzone>
+      <FileUpload.Trigger>Choose file(s)</FileUpload.Trigger>
+      <FileUpload.ItemGroup>
+        <FileUpload.Context>
+          {({ acceptedFiles }) =>
+            acceptedFiles.map((file) => (
+              <FileUpload.Item file={file} key={file.name}>
+                <FileUpload.ItemPreview type="image/*">
```

Comment:
This is the official code from the ark library [ref](https://ark-ui.com/react/docs/components/file-upload) (the first example). It is giving me `ts` error continuously and I could not fix it.
Here is the error:
```
. dev: src/file-uploader/FileUploader.tsx(11,15): error TS2339: Property 'acceptedFiles' does not exist on type 'UseFileUploadContext'.
. dev: src/file-uploader/FileUploader.tsx(12,32): error TS7006: Parameter 'file' implicitly has an 'any' type.
. dev: src/file-uploader/FileUploader.tsx(13,32): error TS2322: Type '{ children: Element[]; file: any; key: any; }' is not assignable to type 'IntrinsicAttributes & FileUploadItemProps & RefAttributes<HTMLLIElement>'.
. dev:   Property 'file' does not exist on type 'IntrinsicAttributes & FileUploadItemProps & RefAttributes<HTMLLIElement>'.
```
Not sure if the library has strict typescript support or not. Can you help me out here? If this does not work out, I can go back to use `react-dropzone`.

Reply:
this is a weird quirk of pnpm - @toha993 faced a similar issue in #113

basically you also need to explicitly add the `@zag-js/file-upload` package

```diff
--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -36,6 +36,7 @@
     "@radix-ui/react-tooltip": "^1.0.7",
     "@reach/auto-id": "^0.18.0",
     "@vanilla-extract/sprinkles": "^1.6.2",
+    "@zag-js/file-upload": "^0.58.3",
     "clsx": "^2.1.1",
     "inter-ui": "^4.0.2"
   },
```

Reply:
hmm i'm finding a lot of weirdness with the ark-ui package

let's switch back to the simpler `react-dropzone` but use ark-ui's API and use a similar one in ours